pkg/apis/events/v1/conversion.go

```diff
@@ -54,5 +63,34 @@ func Convert_core_Event_To_v1_Event(in *k8s_api.Event, out *v1.Event, s conversi
 	out.DeprecatedFirstTimestamp = in.FirstTimestamp
 	out.DeprecatedLastTimestamp = in.LastTimestamp
 	out.DeprecatedCount = in.Count
+
+	// clear DeprecatedSource fields if they match the ReportingController field
+	if out.DeprecatedSource.Component == out.ReportingController {
+		out.DeprecatedSource.Component = ""
+	}
+
 	return nil
 }
+
+func AddFieldLabelConversionsForEvent(scheme *runtime.Scheme) error {
+	return scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.WithKind("Event"),
```

Comment:
this is responsible for matching field selector keys in the events/v1 API to fieldset keys set by `event/strategy.go#SelectableFields`... use `regarding` instead of `involvedObject`, and `reportingController` instead of `source`